File: internal/core/id.go
package core

import (
	"crypto/rand"
	"fmt"
)

// GenerateID returns a 6-character lowercase hex string (e.g., "a1b2c3").
// Uses crypto/rand for cryptographically secure randomness.
func GenerateID() (string, error) {
	b := make([]byte, 3) // 3 bytes = 6 hex chars
	if _, err := rand.Read(b); err != nil {
		return "", fmt.Errorf("generating id: %w", err)
	}
	return fmt.Sprintf("%x", b), nil
}

// IsUniqueID checks whether the given id is not already present in the existing set.
func IsUniqueID(id string, existing map[string]bool) bool {
	return !existing[id]
}

File: src/core/id.ts
import { randomBytes } from 'crypto';

/**
 * Generates an 8-character lowercase hex ID using 4 cryptographically random bytes.
 */
export function generateId(): string {
  return randomBytes(4).toString('hex');
}

/**
 * Checks whether an ID is unique (not present in the existing set).
 * The check is exact and case-sensitive.
 */
export function isUniqueId(id: string, existing: Set<string>): boolean {
  return !existing.has(id);
}
